activetable.js

```diff
@@ -5,12 +5,35 @@ var ActiveTable = (function () {
     NUMERIC = 1;
     STRING = 2;
 
+    // Helper function to determine the number of significant digits in a
+    // floating point literal.  Significant digits are counted from the first
+    // non-zero digit specified, but include trailing zeros.  The string s is
+    // assumed to be a valid floating-point literal and is not validated.
+    function significant_digits(s) {
+        // Regular expression to capture the digits before and after the decimal
+        // point, but not the digits in the exponent.
+        var re = /([0-9]*)\.([0-9]+)|([0-9]+)/;
```

Comment:
@smarnach this regex appears a bit wrong - it has three capturing groups, as a result, there's always one or two items in `re.exec(s).slice(1)` undefined. I suggest using:

    var re = /([0-9]*)\.?([0-9]*)/

You might want to use [regex analyzer](http://www.myezapp.com/apps/dev/regexp/show.ws) to see the difference more clearly.

There's also one subtle difference: if only `.` is passed, original regex will cause an exception to be thrown, while suggested will just return zero. This is not a problem though, since you've declared this function should be feed with sane inputs only, but still nice to be robust in case something goes wrong later.